src/gfx/shaders/Uber_frag.glsl

```diff
@@ -394,38 +383,35 @@ float unpackRGBAToDepth( const in vec4 v ) {
           ) * ( 1.0 / 9.0 );
       #endif
 
-      #ifdef SHADOWMAP_PCF_SOFT
+      #ifdef SHADOWMAP_PCF_RAND
         vec2 texelSize = vec2( 1.0 ) / shadowMapSize;
 
-        float dx0 = - texelSize.x * shadowRadius;
-        float dy0 = - texelSize.y * shadowRadius;
-        float dx1 = + texelSize.x * shadowRadius;
-        float dy1 = + texelSize.y * shadowRadius;
-
-        shadow = (
-        	texture2DShadowLerp( shadowMap, shadowMapSize, shadowCoord.xy + vec2( dx0, dy0 ), shadowCoord.z ) +
-        	texture2DShadowLerp( shadowMap, shadowMapSize, shadowCoord.xy + vec2( 0.0, dy0 ), shadowCoord.z ) +
-        	texture2DShadowLerp( shadowMap, shadowMapSize, shadowCoord.xy + vec2( dx1, dy0 ), shadowCoord.z ) +
-        	texture2DShadowLerp( shadowMap, shadowMapSize, shadowCoord.xy + vec2( dx0, 0.0 ), shadowCoord.z ) +
-        	texture2DShadowLerp( shadowMap, shadowMapSize, shadowCoord.xy, shadowCoord.z ) +
-        	texture2DShadowLerp( shadowMap, shadowMapSize, shadowCoord.xy + vec2( dx1, 0.0 ), shadowCoord.z ) +
-        	texture2DShadowLerp( shadowMap, shadowMapSize, shadowCoord.xy + vec2( dx0, dy1 ), shadowCoord.z ) +
-        	texture2DShadowLerp( shadowMap, shadowMapSize, shadowCoord.xy + vec2( 0.0, dy1 ), shadowCoord.z ) +
-        	texture2DShadowLerp( shadowMap, shadowMapSize, shadowCoord.xy + vec2( dx1, dy1 ), shadowCoord.z )
-        ) * ( 1.0 / 9.0 );
+        vec4 vUv = ((projectionMatrix * vec4(vViewPosition, 1.0)) + 1.0) / 2.0;
+        vec2 vUvNoise = vUv.xy / srcTexelSize * noiseTexelSize;
+
+        vec2 randN = texture2D(noiseTex, vUvNoise).rg;
+        vec2 noiseVec = normalize(randN);
```

Comment:
Is the variable required? Why not to write `noiseVec = normalize(texture2D(noiseTex, vUvNoise).rg);`

Reply:
Done